### PR TITLE
ci: move small-size E2E job to use `e2e-ci.sh`

### LINKED ISSUE #2429: Move small CI job use `e2e-ci.sh`
Currently, the small CI job uses `basic-workflow-tests.sh` - we should create a small section in the `e2e-ci.sh` script and move the job to use that script

### PR DESCRIPTION
**Pull request by nathan-weinberg**
**Issue resolved by this Pull Request:**
Resolves #2429

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [x] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Functional tests have been added, if necessary.
- [x] E2E Workflow tests have been added, if necessary.

### CODE REVIEW

**Review by danmcp — APPROVED** [review]

File: scripts/e2e-ci.sh [hunk 1]
```diff
@@ -362,7 +394,10 @@ test_exec() {
     wait_for_server shutdown $PID
 
     # evaluate tests
-    test_evaluate
+    # note we do not run eval on the small runner due to a lack of GPU resources
+    if [ "$SMALL" -eq 0 ]; then
```
Comment (alimaredia) [inline scripts/e2e-ci.sh:398]: Have you tried to run evaluation on a small runner? Running MT-Bench should be possible with .gguf models, and if you have enough VRAM you could also run MMLU as well.

**Review by alimaredia — APPROVED** [review]
Approving, but I would like to see a follow up issue to see what kind of eval we could test on a small T-shirt size job.

**Comment by nathan-weinberg** [comment]
> Approving, but I would like to see a follow up issue to see what kind of eval we could test on a small T-shirt size job.

I can run some experiments and see about expanding the coverage!

File: scripts/e2e-ci.sh (hunk 1 above)
Comment (nathan-weinberg) [reply scripts/e2e-ci.sh:398]: MMLU didn't work for me with Granite safetensors (not enough VRAM) but I can def try MT Bench with GGUFs and see if we can add it as a followup

**Pull request merged**